yt/frontends/parthenon/fields.py

```diff
@@ -0,0 +1,225 @@
+        # New field naming (i.e., "variable_component") of primitive variables
+        ("prim_density", (rho_units, ["density"], None)),
+        ("prim_velocity_1", (vel_units, ["velocity_x"], None)),
+        ("prim_velocity_2", (vel_units, ["velocity_y"], None)),
+        ("prim_velocity_3", (vel_units, ["velocity_z"], None)),
+        ("prim_pressure", (pres_units, ["pressure"], None)),
+        # Magnetic fields carry units of 1/sqrt(pi) so we cannot directly forward
+        # and need to setup aliases below.
+        ("prim_magnetic_field_1", (mag_units, [], None)),
+        ("prim_magnetic_field_2", (mag_units, [], None)),
+        ("prim_magnetic_field_3", (mag_units, [], None)),
+        # New field naming (i.e., "variable_component") of conserved variables
+        ("cons_density", (rho_units, ["density"], None)),
+        ("cons_momentum_density_1", (mom_units, ["momentum_density_x"], None)),
+        ("cons_momentum_density_2", (mom_units, ["momentum_density_y"], None)),
+        ("cons_momentum_density_3", (mom_units, ["momentum_density_z"], None)),
+        ("cons_total_energy_density", (eng_units, ["total_energy_density"], None)),
+        # Magnetic fields carry units of 1/sqrt(pi) so we cannot directly forward
+        # and need to setup aliases below.
+        ("cons_magnetic_field_1", (mag_units, [], None)),
+        ("cons_magnetic_field_2", (mag_units, [], None)),
+        ("cons_magnetic_field_3", (mag_units, [], None)),
+        # Legacy naming. Given that there's no conflict with the names above,
+        # we can just define those here so that the frontend works with older data.
+        ("Density", (rho_units, ["density"], None)),
+        ("Velocity1", (mom_units, ["velocity_x"], None)),
+        ("Velocity2", (mom_units, ["velocity_y"], None)),
+        ("Velocity3", (mom_units, ["velocity_z"], None)),
+        ("Pressure", (pres_units, ["pressure"], None)),
+        ("MagneticField1", (mag_units, [], None)),
+        ("MagneticField2", (mag_units, [], None)),
+        ("MagneticField3", (mag_units, [], None)),
+        ("MomentumDensity1", (mom_units, ["MomentumDensity1"], None)),
+        ("MomentumDensity2", (mom_units, ["MomentumDensity2"], None)),
+        ("MomentumDensity3", (mom_units, ["MomentumDensity3"], None)),
+        ("TotalEnergyDensity", (eng_units, ["TotalEnergyDensity"], None)),
+    )
+
+    def setup_fluid_fields(self):
+        from yt.fields.magnetic_field import setup_magnetic_field_aliases
+
+        unit_system = self.ds.unit_system
+        # Add velocity fields (if only momemtum densities are given)
+        for i, comp in enumerate(self.ds.coordinates.axis_order):
+            # Support both current and legacy scheme
+            for mom_field_name in ["MomentumDensity", "cons_momentum_density_"]:
+                mom_field = ("parthenon", f"{mom_field_name}{i+1}")
+                if mom_field in self.field_list:
```

Comment:
I updated quite a bit of the field logic (moving things to `known_other_fields` above) and am now wondering if the registered/resolved fields are potentially already available here.
This would simplify logic here (and at several places below) wrt to just checking if `("gas", "momentum_density_x")` is known rather than lookups for our own names in `self.field_list`.